src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunItem.java

```diff
@@ -208,52 +224,55 @@ public void register(boolean slimefun) {
+			SlimefunStartup.getItemCfg().setDefaultValue(id + ".allow-disenchanting", disenchantable);
+			SlimefunStartup.getItemCfg().setDefaultValue(id + ".required-permission", permission);
+			if (keys != null && values != null) {
+				for (int i = 0; i < keys.length; i++) {
+					SlimefunStartup.getItemCfg().setDefaultValue(id + "." + keys[i], values[i]);
 				}
 			}
 			
 			for (World world: Bukkit.getWorlds()) {
 				SlimefunStartup.getWhitelist().setDefaultValue(world.getName() + ".enabled", true);
-				SlimefunStartup.getWhitelist().setDefaultValue(world.getName() + ".enabled-items." + this.name, true);
+				SlimefunStartup.getWhitelist().setDefaultValue(world.getName() + ".enabled-items." + id, true);
 			}
 			
-			if (this.isTicking() && !SlimefunStartup.getCfg().getBoolean("URID.enable-tickers")) {
-			    this.state = State.DISABLED;
+			if (ticking && !SlimefunStartup.getCfg().getBoolean("URID.enable-tickers")) {
+			    state = State.DISABLED;
 			    return;
 			}
 			
-			if (SlimefunStartup.getItemCfg().getBoolean(this.name + ".enabled")) {
+			if (SlimefunStartup.getItemCfg().getBoolean(id + ".enabled")) {
 				if (!Category.list().contains(category)) category.register();
 				
-				this.state = State.ENABLED;
+				state = State.ENABLED;
 				
-				this.replacing = SlimefunStartup.getItemCfg().getBoolean(this.name + ".can-be-used-in-workbenches");
-				this.enchantable = SlimefunStartup.getItemCfg().getBoolean(this.name + ".allow-enchanting");
-				this.disenchantable = SlimefunStartup.getItemCfg().getBoolean(this.name + ".allow-disenchanting");
+				replacing = SlimefunStartup.getItemCfg().getBoolean(id + ".can-be-used-in-workbenches");
+				hidden = SlimefunStartup.getItemCfg().getBoolean(id + ".hide-in-guide");
+				enchantable = SlimefunStartup.getItemCfg().getBoolean(id + ".allow-enchanting");
+				disenchantable = SlimefunStartup.getItemCfg().getBoolean(id + ".allow-disenchanting");
+				permission = SlimefunStartup.getItemCfg().getString(id + ".required-permission");
 				items.add(this);
 				if (slimefun) vanilla++;
-				map_name.put(this.getName(), this.getURID());
-				this.create();
+				map_id.put(id, urid);
+				create();
 				for (ItemHandler handler: itemhandlers) {
 					Set<ItemHandler> handlerset = getHandlers(handler.toCodename());
 					handlerset.add(handler);
 					handlers.put(handler.toCodename(), handlerset);
 				}
 				
-				if (SlimefunStartup.getCfg().getBoolean("options.print-out-loading")) System.out.println("[Slimefun] Loaded Item \"" + this.getName() + "\"");
+				if (SlimefunStartup.getCfg().getBoolean("options.print-out-loading")) System.out.println("[Slimefun] Loaded Item \"" + this.getID() + "\"");
 			} else {
-			    if (this instanceof VanillaItem) this.state = State.VANILLA;
-			    else this.state = State.DISABLED;
+			    if (this instanceof VanillaItem) state = State.VANILLA;
+			    else state = State.DISABLED;
```

Comment:
Do not remove all the 'this.' statements, they are not needed but make it cleaner, clarifying it is a class level variable.

Reply:
Gotcha :)